File: demo-android/app/src/androidTest/java/com/example/demo/e2e/interaction/page/LanguageSwitcherDropdown.kt
package com.example.demo.e2e.interaction.page

import androidx.compose.ui.test.junit4.AndroidComposeTestRule
import com.example.demo.MainActivity

class LanguageSwitcherDropdown(
    rule: AndroidComposeTestRule<*, MainActivity>,
) : ComposePage(rule) {
    fun dropdown() = node("language-switcher")

    fun englishOption() = node("language-option-en")

    fun spanishOption() = node("language-option-es")
}

File: demo-ios/DemoUITests/Interaction/Page/LanguageSwitcherDropdown.swift
import XCTest

final class LanguageSwitcherDropdown: XCUIPage {
    func dropdown() -> XCUIElement {
        app.buttons["language-switcher"]
    }

    func englishOption() -> XCUIElement {
        app.buttons["language-option-en"]
    }

    func spanishOption() -> XCUIElement {
        app.buttons["language-option-es"]
    }
}
